90zfsbootmenu/module-setup.sh

```diff
@@ -121,11 +111,7 @@ install() {
   fi
 
   # Synchronize initramfs and system hostid
-  AA=$(hostid | cut -b 1,2)
-  BB=$(hostid | cut -b 3,4)
-  CC=$(hostid | cut -b 5,6)
-  DD=$(hostid | cut -b 7,8)
-
+  HOSTID="$( hostid )"
   # shellcheck disable=SC2154
-  echo -ne "\\x${DD}\\x${CC}\\x${BB}\\x${AA}" > "${initdir}/etc/hostid"
+  echo -ne "\\x${HOSTID:6:2}\\x${HOSTID:4:2}\\x${HOSTID:2:2}\\x${HOSTID:0:2}" > "${initdir}/etc/hostid"
```

Comment:
This is reasonable, if bash-only; we've already embraced bashisms like arrays where needed and it saves us the `cut`. I assume you've double-checked the ordering, even though it looks correct from here.

Reply:
I tested it by moving /etc/hostid out of the way, creating a new EFI bundle and setting `spl_hostid` on the command line / booting the bundle. The pool was able to be imported so it seems accurate. I can do another check tonight to make sure I'm remembering things correctly.